processor/src/chiplets/hasher/mod.rs

```diff
@@ -76,13 +89,6 @@ mod tests;
 pub struct Hasher {
     trace: HasherTrace,
     aux_trace: AuxTraceBuilder,
-    // TODO: Investigate optimization options, since these lookups are also stored in the bus.
-    // 1. HasherLookup can be lightened to reduce the cost by removing the state from it and looking
-    //    it up in the execution trace when the lookup values are computed and to b_chip.
-    // 2. The Hasher could "provide" lookups immediately instead of storing them and providing them
-    //    during `fill_trace`.
```

Comment:
Following this approach means we don't want to send the Hasher lookups to the chiplets bus during `fill_trace`. Instead, we want the bus to already have all of those lookups in its response rows, so we should provide lookups to the bus at the same time we send requests to the bus